grakn-graql/src/test/java/ai/grakn/graql/internal/reasoner/ReasoningTests.java

```diff
@@ -281,13 +288,21 @@ public void transRelationWithEntityGuardsAtBothEnds() {
         assertEquals(answers.size(), 1);
     }
 
-    @Ignore // TODO: Fix (Bug #16195)
     @Test //Expected result: The query should return a unique match
     public void transRelationWithRelationGuardsAtBothEnds() {
         QueryBuilder qb = testSet11.graph().graql().infer(true);
         String queryString = "match (role1:$x, role2:$y) isa relation3;";
-        QueryAnswers answers = queryAnswers(qb.parse(queryString));
-        assertEquals(answers.size(), 1);
+
+        System.out.println("rel1: ");
```

Comment:
With these annoying tests I usually go for like 10 retests on travis. Maybe try that? After ten retries you can poke me and I will merge. Just a suggestion though.